### DISCUSSION TITLE
about AddRedis

### DISCUSSION THREAD

**Discussion by 1996-Chinese-Chen** [microsoft/aspire#2045] [Q&A] [answered]
Why can’t I specify the redis connection string in AddRedis after the update?

**Comment by davidfowl** [comment, marked as answer]
~Can you provide more details?~ Ah you mean AddRedis(name, connectionString). We removed these overloads and are reworking the how connection strings work. It'll change in the next 2 previews.

For now you can do this:

```C#
var cs = new ConnectionString("redis");

builder.AddProject<P1>("api").WithReference(cs);
```

It'll look for a connection string in configuration.

**Reply by 1996-Chinese-Chen** [reply]
In 8.0.0-preview.1.23557.2 version of AddRedis, there is a parameter called connectionString that can configure the connection string. It will be useless in the next version.And I configure StackExchange .Redis.ConnectionString and I can't change the connection string, if I already have a Redis container, I can't use ，Unless I implement IResourceWithConnectString myself

**Reply by 1996-Chinese-Chen** [reply]
thank you reply
